[EHN,CLN] 공지사항 게시판 crud(+멤버 서비스클래스 삭제)

kimhanui opened the pull request:
EHN :

- 공지사항 게시판 crud 구현

- data.sql 수정(한글허용)

- 공지 최초 생성된 시간 , 수정된 시간 기록 칼럼 있음.

- admin계정으로만 공지 등록하도록 securityConfig에서 url 권한설정

- restController에 글 추가하는 url :  /notice/create로 수정

CLN :

- MemberService, MemberServiceInpl 안쓰는 클래스이므로 삭제.

# 기승전 부탁 
@riyenas0925 

notice.js좀 만들어주실...? js 함수 모아넣고싶은데 잘 안되네
- noticeDetals.jsp, noticeList.jsp에 쓰이는 js함수들을 notice.js에 넣고 정리하고 싶었는데 잘 안됨.
   notice.js에서 jquery를 써서 그런가?
- 지금 있는 notice.js는 noticeDetails.jsp랑 noticeList.jsp에 있는 js함수들 걍 다 모아넣은거라(완전히 일치하진 않음) 다시 만들어야될 것 같어
<hr/>
@2kyung19 

공지사항 url달아놓음( 필요할진 모르겟는데 url 뭐 줘야되는지 잘 몰라서 )
![image](https://user-images.githubusercontent.com/30483337/74368588-507ff980-4e17-11ea-8a16-356b83e5e662.png)

관리자로 록인햇을때 글목록페이지
![image](https://user-images.githubusercontent.com/30483337/74368441-fda64200-4e16-11ea-8901-53cd255fd7d4.png)

관리자로 록인햇을 때 수정가능하게 보임
![image](https://user-images.githubusercontent.com/30483337/74368208-840e5400-4e16-11ea-9504-da507979a27c.png)

일반사용자로 록인
![image](https://user-images.githubusercontent.com/30483337/74368695-80c79800-4e17-11ea-9da2-af152d4e56ff.png)

![image](https://user-images.githubusercontent.com/30483337/74368724-8d4bf080-4e17-11ea-9d1f-20df06918448.png)



Resolved #136

kimhanui committed:
[EHN,CLN] 공지사항 게시판 crud, 멤버 서비스클래스 삭제

EHN :

- 공지사항 게시판 crud 구현

- data.sql 수정(한글허용)

- 공지 최초 생성된 시간 , 수정된 시간 기록 칼럼 있음.

CLN :

- MemberService, MemberServiceInpl 안쓰는 클래스이므로 삭제.

[EHH-2] 공지사항 게시판 crud

EHN :

- admin계정으로만 공지 등록하도록 securityConfig 설정

- restController에 글 추가 url /notice/create로 수정

riyenas0925 commented on src/main/java/kr/ac/sejong/service/NoticeService.java:
```diff
@@ -0,0 +1,62 @@
+public class NoticeService {
+
+    @Inject
+    NoticeRepository repo;
```
@kimhanui @Inject 어노테이션 대신에 생성자를 만들어서 하는 방식으로 주입하자

kimhanui replied:
```java
NoticeRepository repo = new NoticeRepositoryIml();
```
이렇게 하자는거야? 왜 굳이 새로운 객체를 만드는거야? @riyenas0925

riyenas0925 replied:
@kimhanui 아 요게 아니라 private final NoticeRepository repo 이렇게 하고 위에 어노테이션에 @Requireargsconstruct 이거 붙여주면 생성자가 만들어져서 생성자로 주입할수 있게 되거든 

참고 : https://leejisoo860911.tistory.com/m/2

riyenas0925 commented on src/main/java/kr/ac/sejong/service/NoticeService.java:
```diff
@@ -0,0 +1,62 @@
+        return id;
+    }
+
+    public NoticeResponseDto findById(Long id) {
```
@kimhanui 이부분도 @transaction read only로 설정해서 사용 부탁

riyenas0925 reviewed (changes requested):
위에 내용만 수정하고 커밋 해주라 
Pr올라온거에 대해서는 리베이스 할필요 없으

riyenas0925 commented:
@kimhanui 아그리고 코멘트 달린 내용중에 해결 된게 있으면 Resolve conversation 눌러주라

kimhanui committed:
[ENH-3] conversation Resolved

# NoticeService.java :

-  @RequiredArgsConstructor //추가
   public class NoticeService {

   private final NoticeRepository repo; //변경

-  findById, findAllDesc -> @Transactional(readOnly = true)로 설정
